src/main/java/com/google/cloud/tools/managedcloudsdk/install/Downloader.java

```diff
@@ -48,24 +44,40 @@
     this.messageListener = messageListener;
   }
 
-  /** Download and return a {@link Path} to downloaded archive. */
-  @Override
-  public Path call() throws IOException, InterruptedException {
+  /**
+   * Download and return a {@link Path} to downloaded archive, this will overwrite a previously
+   * existing file.
+   */
+  public Path download() throws IOException, InterruptedException {
     if (!Files.exists(destinationFile.getParent())) {
       Files.createDirectories(destinationFile.getParent());
     }
 
+    if (Files.exists(destinationFile) && !Files.isRegularFile(destinationFile)) {
+      throw new IOException("Cannot overwrite non-regular file: " + destinationFile);
+    }
+
     URLConnection connection = address.openConnection();
     connection.setRequestProperty("User-Agent", userAgentString);
 
-    try (BufferedOutputStream out =
-        new BufferedOutputStream(
-            Files.newOutputStream(destinationFile, StandardOpenOption.CREATE_NEW))) {
-      try (InputStream in = connection.getInputStream()) {
-        messageListener.message("Downloading " + address);
+    try (InputStream in = connection.getInputStream()) {
+      long contentLength = connection.getContentLengthLong();
+
+      if (Files.isRegularFile(destinationFile)) {
+        if (Files.size(destinationFile) == contentLength) {
```

Comment:
what if it has same length and different content?

Reply:
Yeah this was initially supposed to get checksums but we don't have that. Maybe for now I'll just redownload.